lib/connection-manager.js

```diff
@@ -1,56 +1,18 @@
-/* globals process */
-'use strict';
-var debug = require('debug')('logdna:lib:connection-manager');
-var log = require('./log');
-var spawn = require('child_process').spawn;
-var os = require('os');
-var url = require('url');
-var fileUtils = require('./file-utilities');
-var winUtils = require('./windows-utilities');
-var apiClient = require('./api-client');
-var linebuffer = require('./linebuffer');
-var WebSocket = require('./logdna-websocket');
-var getProxyFromURI = require('./getProxyFromURI');
-var HttpsProxyAgent = require('https-proxy-agent');
-
-function updateSelf() {
-    if (os.platform() !== 'win32') {
-        // update self
-        return spawn('/bin/bash', ['-c'
-            , 'if [[ ! -z $(which apt-get) ]]; then apt-get update; apt-get install -y --force-yes logdna-agent; elif [[ ! -z $(which yum) ]]; then yum clean expire-cache; yum -y install logdna-agent; elif [[ ! -z $(which zypper) ]]; then zypper refresh; zypper install -y logdna-agent; fi; sleep 1; /etc/init.d/logdna-agent start'
-        ], {
-            detached: true
-        });
-    } else {
-        // update self
-        return spawn('cmd.exe', ['/c', 'choco', 'upgrade', 'logdna-agent', '-y', '--allowunofficial'],
-            {
-                detached: true
-            });
-    }
-}
-
-function restartSelf() {
-    if (os.platform() === 'linux') {
-        // restart self
-        return spawn('/bin/bash', ['-c'
-            , '/etc/init.d/logdna-agent restart'
-        ], {
-            detached: true
-        });
-    } else if (os.platform() === 'win32') {
-        // restart self
-        return spawn('cmd.exe', ['/c'
-            , 'nssm'
-            , 'restart'
-            , 'logdna-agent'
-        ], {
-            detached: true
-        });
-    }
-}
-
-module.exports.connectLogServer = function(config, programName) {
+// External Modules
+const async = require('async');
+const debug = require('debug')('logdna:lib:connection-manager');
+const os = require('os');
+const url = require('url');
+const HttpsProxyAgent = require('https-proxy-agent');
+
+// Internal Modules
+const fileUtils = require('./file-utilities');
+const winUtils = require('./windows-utilities');
+const linebuffer = require('./linebuffer');
+const WebSocket = require('./logdna-websocket');
```

Comment:
Same as Mike's comment below - I think the point is to remove websocket, get rid of the reference, get rid of this line, delete `logdna-websocket.js`.